new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+            logger.info(item);
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaces");
+      }
+
+      return doesnotExist;
+    };
+
+  }
+
+  /**
+   * Deletes artifacts in the Kubernetes cluster in the given namespace.
+   *
+   * @param namespace name
+   */
+  public static void deleteNamespacedArtifacts(String namespace) {
+    logger.info("Deleting artifacts in namespace {0}", namespace);
+
+    // Delete all Domain objects in given namespace
+    try {
+      for (var item : Kubernetes.listDomains(namespace).getItems()) {
+        Kubernetes.deleteDomainCustomResource(item.getMetadata().getName(), namespace);
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete domains");
+    }
+
+    // Delete replicasets
+    try {
+      for (var item : Kubernetes.listReplicaSets(namespace).getItems()) {
+        Kubernetes.deleteReplicaSet(namespace, item.getMetadata().getName());
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete replica sets");
+    }
+
+    // Delete jobs
+    try {
+      for (var item : Kubernetes.listJobs(namespace).getItems()) {
+        Kubernetes.deleteJob(namespace, item.getMetadata().getName());
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete jobs");
+    }
+
+    // Delete configmaps
+    try {
+      for (var item : Kubernetes.listConfigMaps(namespace).getItems()) {
+        Kubernetes.deleteConfigMap(item.getMetadata().getName(), namespace);
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete config maps");
+    }
+
+    // Delete secrets
+    try {
+      for (var item : Kubernetes.listSecrets(namespace).getItems()) {
+        Kubernetes.deleteSecret(item.getMetadata().getName(), namespace);
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete secrets");
+    }
+
+    // Delete pv
+    try {
+      for (var item : Kubernetes.listPersistentVolumeClaims(namespace).getItems()) {
+        String label = Optional.ofNullable(item)
+            .map(pvc -> pvc.getMetadata())
+            .map(metadata -> metadata.getLabels())
+            .map(labels -> labels.get("weblogic.domainUid")).get();
+        for (var pv : Kubernetes.listPersistentVolumes(
+            String.format("weblogic.domainUid = %s", label)).getItems()) {
+          Kubernetes.deletePv(pv.getMetadata().getName());
+        }
+      }
+    } catch (ApiException ex) {
+      logger.warning(ex.getResponseBody());
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Failed to delete persistent volumes");
+    }
+
+    // Delete deployments
```

Comment:
probably should do deployments before pvcs, config maps, secrets, rs's, otherwise those could fail

Reply:
moved the deployments delete above  pvcs, config maps, secrets, rs's,